src/mmda/types/document.py

```diff
@@ -160,37 +159,61 @@ def _annotate_box_group(
 
         all_page_tokens = dict()
         derived_span_groups = []
+        token_box_in_box_group = None
 
         for box_id, box_group in enumerate(box_groups):
 
-            all_token_spans_with_box_group = []
+            all_tokens_overlapping_box_group = []
 
             for box in box_group.boxes:
 
                 # Caching the page tokens to avoid duplicated search
                 if box.page not in all_page_tokens:
-                    cur_page_tokens = all_page_tokens[box.page] = list(
-                        itertools.chain.from_iterable(
-                            span_group.spans
-                            for span_group in self.pages[box.page].tokens
+                    cur_page_tokens = all_page_tokens[box.page] = self.pages[
+                        box.page
+                    ].tokens
+                    if token_box_in_box_group is None:
+                        # Determine whether box is stored on token SpanGroup span.box or in the box_group
+                        token_box_in_box_group = all(
+                            [
+                                (
+                                    (hasattr(token.box_group, "boxes") and len(token.box_group.boxes) == 1)
```

Comment:
I think this might be a little off. `SpanGroup.box_group` is `Optional[BoxGroup]`, and `BoxGroup.boxes` is `List[Box]`.

So this should probably be:

```python
(hasattr(token, "box_group") and len(token.box_group.boxes) == 1)
```

Reply:
If we don't already have it, we could use a test case where we don't have box groups on the tokens, which would throw an exception as written right now.